streaming/receiver.c

```diff
@@ -95,6 +95,54 @@ PARSER_RC streaming_timestamp(char **words, void *user, PLUGINSD_ACTION *plugins
     return PARSER_RC_ERROR;
 }
 
+#define CLAIMED_ID_MIN_WORDS 3
+PARSER_RC streaming_claimed_id(char **words, void *user, PLUGINSD_ACTION *plugins_action)
+{
+    UNUSED(user);
+    UNUSED(plugins_action);
+
+    int i;
+    uuid_t uuid;
+
+    for (i = 0; words[i]; i++) ;
+    if (i != CLAIMED_ID_MIN_WORDS) {
+        error("Command CLAIMED_ID came malformed %d parameters are compulsory", CLAIMED_ID_MIN_WORDS - 1);
+        return PARSER_RC_ERROR;
+    }
+
+    // We don't need to parsed UUID
+    // just do it to check the format
+    if(uuid_parse(words[1], uuid)) {
+        error("The host GUID received doesn't have format of valid UUID!");
+        return PARSER_RC_ERROR;
+    }
+    if(uuid_parse(words[2], uuid) && strcmp(words[2], "NULL")) {
+        error("The Claim ID received doesn't have format of valid UUID!");
+        return PARSER_RC_ERROR;
+    }
+
+    RRDHOST *host = rrdhost_find_by_guid(words[1], 0);
+    if (!host) {
+        error("Host with GUID %s not found. Ignoring", words[1]);
+        return PARSER_RC_OK; //the message is OK problem must be somewehere else
+    }
+
+    if (host == localhost) {
+        error("Streaming Child attempted to change claiming id of parent (me). This is not allowed.");
+        return PARSER_RC_OK;
+    }
+
+    netdata_mutex_lock(&host->claimed_id_lock);
+    if (host->claimed_id)
+        freez(host->claimed_id);
+    host->claimed_id = strcmp(words[2], "NULL") ? strdupz(words[2]) : NULL;
+    netdata_mutex_unlock(&host->claimed_id_lock);
+
+    rrdpush_claimed_id(host);
+
+    return PARSER_RC_OK;
+}
```

Comment:
The error messages in this function are not useful. If this breaks then we need to diagnose it by checking logs, so at the point that an error is encountered there needs to be enough information in the log to pinpoint the problem.
* If the claim ID string is not a valid form then show the string.
* If the GUID is not valid then show the string.
* If the child is attempting to change the wrong UUID then show the incorrect UUIDS.
This does not jut help us - users with broken config will look in their logs to try to understand the problem and it should contain enough information.

The conversation above has been marked resolved - but the problem is not resolved. A child cannot change the claim id of an arbitrary host. This means that configuration errors propagate through the system and break the state on other nodes. Find another way to handle it,

Reply:
I will improve the logging. 
Host must be able to modify claim_id of another host than self (as it can itself have children). Not sure yet how to improve this other than adding visibility of structure of children below (as a tree instead of list)

Reply:
The terminology is a bit confusing here (historically I think the terms have shifted a bit). When a sender connects to a receiver it only transmits a single *host* - in this context the host is a particular database. When a proxy node is transmitting the data for multiple children to a parent there one stream per host. So the protocol that is coming over the wire is only describing a single host (even though the node may be streaming multiple hosts over multiple connections).

Reply:
ah I see what you mean now. Knew there is connection per host but somehow misunderstood/didn't make a connection with what you try to say.

Changed it. Thanks